### PR TITLE
Simplify namespace selector for ResourceRefNamespaced

### PR DESCRIPTION
**Pull request by fabritsius**

### CODE REVIEW

File: pkg/api/v1/project/integration.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ func (i Integration) ToAtlas(c client.Client, defaultNS string) (result *mongodb
 			return
 		}
 
-		*target, err = passwordField.ReadPassword(c, getNamespace(passwordField.Namespace, defaultNS))
+		*target, err = passwordField.ReadPassword(c, defaultNS)
```
Comment (Sugar-pack) [inline pkg/api/v1/project/integration.go:85]: Why is the namespace from the passwordField ignored here?

**Review by Sugar-pack — APPROVED** [review]

File: pkg/api/v1/common/common.go [hunk 2]
```diff
@@ -72,3 +69,15 @@ func (rn *ResourceRefNamespaced) ReadPassword(kubeClient client.Client, parentNa
 	}
 	return "", nil
 }
+
+// SelectNamespace returns first non-empty namespace from the list
+// "", "", "first", "second" => "first"
+func SelectNamespace(namespaces ...string) string {
```
[the remark this answers is not in the document]
Comment (fabritsius) [reply pkg/api/v1/common/common.go:75]: Well, not really, I see your point 😅

I thought about it. My first version of this function had only two inputs like `namespace` and `parentNamespace`. Then I started thinking how do I explain that we prefer one over another and if so which one. How do I name inputs for cases where we decide that we actually want a parent namespace to be overridden.

So I noticed that there is a certain directional thing and this procedure can be generalised. The resulting function can be an `if ... else` for this case but the current version is hardly complicated and a general one.

[the remark this answers is not in the document]
Comment (igor-karpukhin) [reply pkg/api/v1/common/common.go:75]: Got it! I was thinking about maybe something like this 
```suggestion
func targetNamespaceOrDefault(namespace, default string) string {
    if namespace != "" {
        return namespace
    }
    return default
}
```

Up to you though 🙂

**Review by helderjs — APPROVED** [review]

File: pkg/api/v1/common/common.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (fabritsius) [reply pkg/api/v1/common/common.go:75]: Let's keep as is 🙃

**Pull request merged**